src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -81,18 +81,18 @@ public static Dataset datasetRequestToDataset(DatasetRequestModel datasetRequest
                 .assetSpecifications(assetSpecifications);
     }
 
-    private static List<StorageResource> createGcpStorageResourceValues(String providedRegion) {
-        final GoogleRegion region = Optional.ofNullable(providedRegion)
+    public static GoogleRegion getRegionFromDatasetRequestModel(DatasetRequestModel datasetRequestModel) {
+        return Optional.ofNullable(datasetRequestModel.getRegion())
             .map(GoogleRegion::fromValue)
             .orElse(GoogleRegion.DEFAULT_GOOGLE_REGION);
```

Comment:
I'm wondering if this should be in `GoogleRegion` instead, since that's the class that owns the notion of a default region. We already have `GoogleRegion.fromValue()`, so it could be `GoogleRegion.fromValueWithDefault()`. Or you could have `fromValue()` return the default unless a `null` return value is used elsewhere to detect an invalid region.

Reply:
That's a good idea!